### PR TITLE
[quic]impose a limit on the number of connection migrations

### PR DESCRIPTION
**Pull request by RenjieTang**
Commit Message: Limit the max number of connection migrations to 5.
Additional Description: This prevents too many new sockets being created.
Risk Level: Low. Can be turned off via num_timeouts_to_trigger_port_migration proto
Testing: integration test
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: n/a

### CODE REVIEW

File: source/common/quic/envoy_quic_client_connection.h [hunk 1]
```diff
@@ -12,6 +12,8 @@
 namespace Envoy {
 namespace Quic {
 
+constexpr uint8_t kMaxNumSocketSwitches = 5;
```
Comment (RyanTheOptimist) [inline source/common/quic/envoy_quic_client_connection.h:15]: nit: Brief comment, please.

**Review by RyanTheOptimist — COMMENTED** [review]
Modulo a request for a comment, this looks good. But I do have a question. The PR descriptions mentions:

Risk Level: Low. Can be turned off via num_timeouts_to_trigger_port_migration proto

But that turns off connection migration, it does not turn of the new behavior in this PR, right?

**Comment by RenjieTang** [comment]
> Modulo a request for a comment, this looks good. But I do have a question. The PR descriptions mentions:
> 
> Risk Level: Low. Can be turned off via num_timeouts_to_trigger_port_migration proto
> 
> But that turns off connection migration, it does not turn of the new behavior in this PR, right?

Yes, the proto turns off connection migration as a whole. I put it there so that ppl know how to disable this feature in case anything goes wrong.

**Review by RyanTheOptimist — APPROVED** [review]

**Pull request merged**